src/actions/write/saveObservations.ts

```diff
@@ -46,98 +56,121 @@ export const saveObservations = async (
   { caller, input }: PstAction,
 ): Promise<ContractWriteResult> => {
   // get all other relevant state data
-  const { observations, gateways, settings } = state;
-  const { observerReportTxId, failedGateways } = new SaveObservations(input); // does validation on constructor
-  const currentEpochStartHeight = getEpochStart({
-    startHeight: DEFAULT_START_HEIGHT,
-    epochBlockLength: DEFAULT_EPOCH_BLOCK_LENGTH,
-    height: +SmartWeave.block.height,
+  const { observations, gateways, settings, distributions } = state;
+  const { gatewayAddress, observerReportTxId, failedGatewayFQDNs } =
+    new SaveObservations(input, caller);
+
+  // TODO: check if current height is less than epochZeroBlockHeight
+  if (+SmartWeave.block.height < distributions.epochZeroBlockHeight) {
+    throw new ContractError(
+      `Observations cannot be submitted before block height: ${distributions.epochZeroBlockHeight}`,
+    );
+  }
```

Comment:
this is new - it'll throw out interactions that come in before our `epochZeroStartHeight`